compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/ProvidesMethodFactoryGenerator.kt

```diff
@@ -52,8 +53,22 @@ internal class ProvidesMethodFactoryGenerator : CodeGenerator {
         .flatMap { clazz ->
           clazz
               .functions(includeCompanionObjects = true)
-              .asSequence()
               .filter { it.hasAnnotation(daggerProvidesFqName) }
+              .also { functions ->
+                // Check for duplicate function names.
+                val duplicateFunctions = functions
+                    .groupBy { it.requireFqName() }
+                    .filterValues { it.size > 1 }
+
+                if (duplicateFunctions.isNotEmpty()) {
+                  throw AnvilCompilationException(
+                      element = clazz,
+                      message = "Cannot have more than one binding method with the same name in " +
+                          "a single module: ${duplicateFunctions.keys.joinToString()}"
+                  )
+                }
+              }
+              .asSequence()
```

Comment:
Why move the asSequence?

Reply:
Because the outer chain is a sequence and Kotlin 1.3 doesn't have the overloads for lists, yet.

Reply:
Not sure why I moved it down, though. I'll revert that.